code/modules/reagents/reagent_containers/blood_pack.dm

```diff
@@ -17,6 +17,20 @@
 	on_reagent_change()
 		update_icon()
 
+		if(volume == 0 && name != "Empty Bloodback")
+			name = "Empty Bloodpack"
+			desc = "Empty bloodpacks are good in vampire movies, but bad in hospitals."
+		else if (reagents.reagent_list.len > 0)
+			var/target_type = null
+			var/the_volume = 0
+			for(var/datum/reagent/A in reagents.reagent_list)
+				if(A.volume > the_volume && ("blood_type" in A.data))
+					the_volume = A.volume
+					target_type = A.data["blood_type"]
+			name = "[target_type] Bloodpack"

```

Comment:
I suggest you add an alt case if there's no blood in it at all.
I mean, ideally, it shouldn't be called a blood bag unless the largest reagent quantity actually is blood, so you might want to do

    if(target_type)
        name = "[target_type] Bloodpack"
        desc = "A bloodpack filled with [target_type] blood."
    else
        name = "murky Bloodpack"
        desc = "A bloodpack filled with mysterious liquid."